v5.37 Changelog (#4702)
mattermost/docs: source/administration/changelog.md

Also see [changelog in progress](https://bit.ly/2nK3cVf) for the next release.

## Release v5.37 - [Extended Support Release](https://docs.mattermost.com/administration/extended-support-release.html)

**Release Day: 2021-07-16**

### Deprecations
 - The ``platform`` binary and “--platform” flag have been deprecated. If you are using the “--platform” flag or are using the ``platform`` binary directly to run the Mattermost server application via a systemd file or custom script, you will be required to use only the mattermost binary.

### Important Upgrade Notes
 - [Collapsed Reply Threads](https://mattermost.com/blog/collapsed-reply-threads-beta/) are available as beta in Mattermost Server v5.37 and later. It’s expected that you may experience bugs as we stabilize the feature. In particular, please be aware of [the known issues documented here](https://docs.mattermost.com/help/messaging/organizing-conversations.html#known-issues).

>  - v5.37 adds support for emoji standard v13.0. If you have added a custom emoji in the past that uses one of the new system names, then it is going to get overwritten by the system emoji. The workaround is to change the custom emoji name.

Reviewer: ```suggestion
 - v5.37 adds support for emoji standard v13.0. If you have added a custom emoji in the past that uses one of the new system names, then that custom emoji is going to get overwritten by the system emoji. The workaround is to change the custom emoji name.
```

#### Emoji Enhancements with Skin Tone Selection
 - Added support for emoji standard v13.0. Users now have the ability to choose various skin tones using the Mattermost emoji picker. Mobile support is included in v1.45 Mobile App release (July 16th).

#### Improved Enterprise Trial Experience (Enterprise Editions E0, E10, E20)
 - After a Self-Managed trial ends, admins can optionally contact sales or make a purchase in a single click.

#### Focalboard: Grouped Table view, New properties, and More (Beta)
 - Focalboard tables can now be grouped by a property, for example allowing you to quickly see tasks per epic or owner.

#### Incident Collaboration Updates
 - The update includes availability on all editions, playbook keyword monitoring, retrospective report, and playbook dashboard.

#### English-Australian Language Support
 - Mattermost is now available in English-Australian.

### Improvements

#### User Interface (UI)
 - In the at-mention autocomplete, the user’s nickname is no longer shown when (you) is present.
 - Updated the help text on the **Add Users** channel modal.
 - Added the ability to upload ``.jpeg`` files on Linux. Uploading ``.jpg`` files was already supported.

>  - Before anything is typed, a list of channels you are a member of is now shown and sorted by recency in the channel switcher. The number of channels shown in the channel switcher in the default state are capped to maximum of 20. If there is no recent channel activity (in the case of new users), an alphabetical list of channels is shown displaying **My Channels** first and other **Public Channels** next, capped at a maximum of 20 by default.

Reviewer: ```suggestion
 - Before anything is typed, a list of channels you are a member of is now shown and sorted by recency in the channel switcher. The number of channels shown in the channel switcher in the default state are capped to a maximum of 20. If there is no recent channel activity (in the case of new users), an alphabetical list of channels is shown displaying **My Channels** first and other **Public Channels** next, capped at a maximum of 20 by default.
```

>  - Before anything is typed, a list of channels you are a member of is now shown and sorted by recency in the channel switcher. The number of channels shown in the channel switcher in the default state are capped to a maximum of 20. If there is no recent channel activity (in the case of new users), an alphabetical list of channels is shown displaying **My Channels** first and other **Public Channels** next, capped at a maximum of 20 by default.

Reviewer 2: This feels a bit long winded, Propose: "The Channel Switcher now displays recently viewed channels when launched"

 - Polish, German, and Italian languages were downgraded to beta as they are [no longer actively maintained](https://handbook.mattermost.com/contributors/contributors/localization#translation-quality).

>  - Added the ability in the custom status tooltips, status dropdown and profile popover to select an expiry time for custom statuses. Also added new components ``ExpiryMenu`` and ``DateTimeInput`` to the custom status modal.

Reviewer: ```suggestion
 - Added the ability in the custom status tooltips, status dropdown, and profile popover to select an expiry time for custom statuses. Also added new components ``ExpiryMenu`` and ``DateTimeInput`` to the custom status modal.
```

>  - Added the ability in the custom status tooltips, status dropdown, and profile popover to select an expiry time for custom statuses. Also added new components ``ExpiryMenu`` and ``DateTimeInput`` to the custom status modal.

Reviewer 2: Focus on end user benefit: "Custom statuses can now be set to expire after common time intervals or custom selected dates and times.

#### Performance
 - Added some improvements to typing performance.

#### Administration
 - Improved memory performance for large image uploads, particularly PNGs with transparency.
 - Optimized the bulk import process by no longer requiring the server to write the incoming archive to the filesystem when unzipping it.
 - Added channel restore and channel privacy change endpoints to the local mode using the System bot.

### Bug Fixes

>  - Fixed an issue where users were unable to set a custom status emoji via slash command by adding the logic for detecting unicode emoji and setting it as custom status emoji via slash commands.

Reviewer: ```suggestion
 - Fixed an issue where users were unable to set a custom status emoji via slash command. Added the logic for detecting unicode emoji and setting it as a custom status emoji via slash commands.
```

### API Changes
 - Added a new field, team_id, to the response of ``POST api/v4/groups/{group_id}/channels/{channel_id}/link`` to add a team ID to the response when linking a channel to a group.
 - Added an optional ``collapsed_threads_supported`` parameter to /channels/members/{userId}/view to indicate that the client supports collapsed threads.
 - Added an optional ``collapsed_threads_supported`` parameter to /users/{userId}/posts/{postId}/set_unread to indicate that the client supports collapsed threads.
 - Updated the webapp to pass ``collapsed_threads_supported`` parameters to the server to indicate that the webapp supports collapsed reply threads.
 - Updated the webapp to correctly mark channels and threads as unread/read when marking root and reply posts as unread/read.
 - Added a new endpoint ``GET /trial-license/prev`` for fetching last used trial license.
 - Added two new fields in ``CustomStatus`` struct and modified the APIs to validate and handle them.

### Go Version
 - v5.37 is built with Go ``1.15.5``.

### Open Source Components
 - Removed ``reselect`` from https://github.com/mattermost/mattermost-webapp/.

### Known Issues

>  - ``config.json`` can reset when running the command ``systemctl restart mattermost``, and when running any commands that write to the config (e.g. ``config`` or ``plugin``) [MM-33752](https://mattermost.atlassian.net/browse/MM-33752), [MM-32390](https://mattermost.atlassian.net/browse/MM-32390).

Reviewer 2: Let's please add a bullet here at the top linking to the CRT doc known issues section
